applications/MeshingApplication/custom_utilities/tetrahedra10_mesh_converter_utility.h

```diff
@@ -0,0 +1,356 @@
+            this_model_part.Nodes()(i3),
+            this_model_part.Nodes()(i4),
+            this_model_part.Nodes()(i5),
+            this_model_part.Nodes()(i6),
+            this_model_part.Nodes()(i7),
+            this_model_part.Nodes()(i8),
+            this_model_part.Nodes()(i9)
+        );
+        return geom;
+    }
+
+    /**
+    * Creates a new triangle3D6
+    * @return the reference to the new triangle
+    */
+    Triangle3D6<Node<3>> GenerateTriangle3D6(ModelPart& this_model_part, array_1d<int, 6>& node_ids) {
+        unsigned int i0   = node_ids[0];
+        unsigned int i1   = node_ids[1];
+        unsigned int i2   = node_ids[2];
+        unsigned int i3   = node_ids[3];
+        unsigned int i4   = node_ids[4];
+        unsigned int i5   = node_ids[5];
+
+        Triangle3D6<Node<3> > geom(
+                this_model_part.Nodes()(i0),
+                this_model_part.Nodes()(i1),
+                this_model_part.Nodes()(i2),
+                this_model_part.Nodes()(i3),
+                this_model_part.Nodes()(i4),
+                this_model_part.Nodes()(i5)
+                );
+        return geom;
+    }
+
+        /**
+    * It erases the old Tetrahedra3D4 elements and it creates the new Tetrahedra3D10 ones
+    * @param Coord: The compressed matrix containing at (i,j) the id of the node created between nodes i,j
+    * @param New_Elements: The new elements created
+    * @param interpolate_internal_variables: A boolean that defines if it is necessary to interpolate the internal variables
+    * @return this_model_part: The model part of the model (it is the input too)
+    */
+
+    void EraseOldElementAndCreateNewElement(
+            ModelPart& this_model_part,
+            const compressed_matrix<int>& Coord,
+            PointerVector< Element >& NewElements,
+            bool interpolate_internal_variables
+    ) override
+    {
+	ElementsArrayType& rElements = this_model_part.Elements();
+	ElementsArrayType::iterator it_begin = rElements.ptr_begin();
+	ElementsArrayType::iterator it_end = rElements.ptr_end();
+
+	const ProcessInfo& rCurrentProcessInfo = this_model_part.GetProcessInfo();
+	int edge_ids[6];
+	std::vector<int> node_ids;
+
+	for (ElementsArrayType::iterator it = it_begin; it != it_end; ++it)
+	{
+        GlobalPointersVector< Element >& rChildElements = it->GetValue(NEIGHBOUR_ELEMENTS);
+        rChildElements.resize(0);
+
+        CalculateEdges(it->GetGeometry(), Coord, edge_ids, node_ids);
+
+        // Generate the new Tetrahedra3D10 element
+        Tetrahedra3D10<Node<3>> geom = GenerateTetrahedra(this_model_part, node_ids);
+        const Element& rElem = KratosComponents<Element>::Get("Element3D10N");
+        Element::Pointer p_element;
+        p_element = rElem.Create(it->Id(), geom, it->pGetProperties());
+        p_element->Initialize(rCurrentProcessInfo);
+        p_element->InitializeSolutionStep(rCurrentProcessInfo);
+        p_element->FinalizeSolutionStep(rCurrentProcessInfo);
```

Comment:
I assume that you require this in order to do the internal variables interpolation below. However, I'm afraid that in here you might be messing the internal values in constitutive models that are historical.

Shouldn't this be a pure geometrical operation to be done at the beginning of the simulation?

Reply:
I can only imagine that this should be run at the beginning of the stage. (as a modeler) but then elements may have constitutive laws and historical value from previous stages. Something to consider in our multi stage discussion.

Reply:
I also agree with @pooyan-dadvand 's observation. The unique way that comes to my mind to solve this is to do an L2 to projection and then a mortar mapping between the two meshes (note that we'll require to have both meshes at same time).  

This mechanism is definitively out of the scope of this PR so lets leave this as an observation for the future.